### PR TITLE
feat: validates UUID to an existing enterprise customer

### PR DESCRIPTION
**Pull request by brobro10000**
Implements the 'basic_list' enterprise customer API endpoint to dynamically create a dropdown based on the user's inputted value. Also makes use of `debounce` to prevent query on key down, and delays api call to 1 second after the final keystroke is complete.

### CODE REVIEW

File: src/data/services/EnterpriseApiService.js [hunk 1]
```diff
@@ -8,7 +8,11 @@ class LmsApiService {
 
   static enterpriseCatalogQueriesUrl = `${LmsApiService.baseUrl}/enterprise/api/v1/enterprise_catalog_query/`;
 
+  static enterpriseCustomersBasicListUrl = `${LmsApiService.baseUrl}/enterprise/api/v1/enterprise-customer/basic_list/`;
+
   static fetchEnterpriseCatalogQueries = () => LmsApiService.apiClient().get(LmsApiService.enterpriseCatalogQueriesUrl);
+
+  static fetchEnterpriseCustomersBasicList = (query) => LmsApiService.apiClient().get(`${LmsApiService.enterpriseCustomersBasicListUrl}${query !== undefined ? `?q=${query}` : ''}`);
```
Comment (pwnage101) [inline src/data/services/EnterpriseApiService.js]: ditto https://github.com/openedx/edx-enterprise/pull/1742#pullrequestreview-1404735552

File: src/data/services/EnterpriseApiService.js [hunk 2]
```diff
@@ -8,7 +8,11 @@ class LmsApiService {
 
   static enterpriseCatalogQueriesUrl = `${LmsApiService.baseUrl}/enterprise/api/v1/enterprise_catalog_query/`;
 
+  static enterpriseCustomersBasicListUrl = `${LmsApiService.baseUrl}/enterprise/api/v1/enterprise-customer/basic_list/`;
+
   static fetchEnterpriseCatalogQueries = () => LmsApiService.apiClient().get(LmsApiService.enterpriseCatalogQueriesUrl);
+
+  static fetchEnterpriseCustomersBasicList = (query) => LmsApiService.apiClient().get(`${LmsApiService.enterpriseCustomersBasicListUrl}${query !== undefined ? `?name_or_uuid=${query}` : ''}`);
```
Comment (pwnage101) [inline src/data/services/EnterpriseApiService.js]: Nit: also rename the function argument "query" -> "enterprise_name_or_uuid" to avoid all confusion

**Review by pwnage101 — APPROVED** [review]

File: src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const { setCustomerUUID, getCustomers } = useProvisioningContext();
+  const [selected, setSelected] = useState({ title: '' });
+  const [dropdownValues, setDropdownValues] = useState(['No matching enterprise']);
+
+  const debouncedSearch = useMemo(() => debounce(getCustomers, 500, {
+    leading: false,
+  }), [formData.enterpriseUUID]);
+  const handleOnSelected = (value) => {
+    if (value && value.includes('---')) {
+      const valueUuid = value.split(' --- ')[1].trim();
```
Comment (adamstankiewicz) [inline src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx:26]: nit: Is it worth having a comment with a brief explanation for why this `.includes('---')` and `.split(' --- ')` is necessary?

File: src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  };
+  const updateDropdown = (value) => {
+    setCustomerUUID(value);
+    if (value?.length === 0 || customers.length === 0) {
+      return setDropdownValues(['No matching enterprise']);
+    }
+    return updateDropdownList();
+  };
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
```
Comment (adamstankiewicz) [inline src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx:45]: [clarification] Can you explain why this `setTimeout` is necessary?

File: src/data/services/EnterpriseApiService.js (hunk 2 above)
Comment (adamstankiewicz) [reply src/data/services/EnterpriseApiService.js]: Bump.

File: src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      debouncedSearch(formData.enterpriseUUID);
+    }, 500);
+    return () => clearTimeout(delayDebounceFn);
+  }, [formData.enterpriseUUID]);
```
Comment (adamstankiewicz) [inline src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx]: `debouncedSearch` is not in `useEffect` dependency array. With it added, it _may_ resolve the need for the `setTimeout` above.

Comment (brobro10000) [reply src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx]: Did not seem to fix the debounce issue I was facing :(

File: src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx (hunk 4 above)
Comment (brobro10000) [reply src/Configuration/Provisioning/ProvisioningForm/ProvisioningFormCustomerDropdown.jsx:45]: The setTimeout is the time between the onChange effect from the users interaction (typing a uuid/customer name).
Not adding that in (even with the updated dependency array results in a call of debouncedSearch every key stroke after the delay that was set within the debounce function. The timeout is acting as a buffer before the the function is called.

**Pull request merged**